### PR TITLE
[WFLY-12687] Updated Smallrye Config to 1.3.11

### PR DESCRIPTION
**Pull request by tobiasstadler**
Issue: https://issues.jboss.org/browse/WFLY-12687

### CODE REVIEW

**Review by rhusar — CHANGES_REQUESTED** [review]
Tests need to be adapted as well.
```
ConfigProviderTest.testEnvironmentConfigSource
EmptyValuesTest.testEmptyStringProgrammaticLookup
EmptyValuesTest.testEmptyStringPropertyFromConfigFile
```

**Comment by rhusar** [comment]
Also a component upgrade Jira is needed and needs to prefix the commit message.

**Comment by jmesnil** [comment]
@tobiasstadler thanks for the PR.

Your PR looks fine but I'll put it on hold for the time being.
After smallrye-config 1.3.6 was released, changes in the components were made for its use in Quarkus.
They resulted in challenges to the TCK about the `EmptyValuesTest` tests that fail in our WildFly Jobs.

We are in the process of releasing smallrye-config 1.3.10 to run the MP Config TCK with Quarkus. At this point, we will be able to upgrade smallrye-config in WildFly with the same TCK challenges that Quarkus has.

**Review by jmesnil — CHANGES_REQUESTED** [review]
smallrye-config 1.3.11 has been released and this is the version we should upgrade to: https://mvnrepository.com/artifact/io.smallrye/smallrye-config/1.3.11

It challenges some MicroProfile Config TCK tests (as defined in https://github.com/smallrye/smallrye-config/blob/master/testsuite/tck/tck-suite.xml).

We should update https://github.com/wildfly/wildfly/blob/master/testsuite/integration/microprofile-tck/config/tck-suite.xml to add the same exclusions.

@tobiasstadler do you want to take care of ^^? thanks

**Comment by tobiasstadler** [comment]
@jmesnil I updated the PR

**Review by rhusar — CHANGES_REQUESTED** [review]
Tests are failing:

org.eclipse.microprofile.metrics.tck
(1)

GlobalTagsTest.customConfigSource
 

org.wildfly.test.integration.microprofile.config.smallrye.app
(2)

MicroProfileConfigTestCase.testGetWithArraySetListDefaultProperties
 

MicroProfileConfigTestCase.testGetWithArraySetListOverriddenProperties
 

TestSuite: org.eclipse.microprofile.config.tck
(1)

ConfigProviderTest.testEnvironmentConfigSource

**Comment by tobiasstadler** [comment]
@rhusar I am not sure what to with GlobalTagsTest.customConfigSource. As per the api spec `ConfigProviderResolver#registerConfig` should throw an IllegalStateException if there is already a Config registered within the Application (with is the case here). I think the test needs to call `releaseConfig` before  `registerConfig`. What do you think?

File: testsuite/integration/microprofile-tck/config/tck-suite.xml [hunk 1]
```diff
@@ -26,6 +26,19 @@
                     <exclude name=".*" />
                 </methods>
             </class>
+            <!-- These test are also exclueded in the SmallRye TCK -->
```
Comment (rhusar) [inline testsuite/integration/microprofile-tck/config/tck-suite.xml]: s/exclueded/excluded

**Review by rhusar — CHANGES_REQUESTED** [review]

**Comment by rhusar** [comment]
@tobiasstadler Its possible, I would need to have a better look...

**Comment by jmesnil** [comment]
@tobiasstadler same comment than for #12719 

Thanks for your help!

**Pull request closed without merging**